src/codec.rs

```diff
@@ -901,30 +939,9 @@ impl<K: Decode + Ord, V: Decode> Decode for BTreeMap<K, V> {
 }
 
 impl Encode for () {
-	fn encode_to<T: Output>(&self, _dest: &mut T) {
-	}
-
 	fn using_encoded<R, F: FnOnce(&[u8]) -> R>(&self, f: F) -> R {
 		f(&[])
 	}
-
-	fn encode(&self) -> Vec<u8> {
```

Comment:
Why did you remove this?

Reply:
I'm pretty sure it's going to be optimized out by the compiler anyway and just clutters the code. But I'm ok to bring it back if you think it's worth having.

Reply:
I would really like to see to get `Vec::new` back. I'm not 100% sure that the compiler will optimize this away. Could be, but involves some function calls that he would need to optimize into a single `Vec::new()`.